pkg/app/api/grpcapi/piped_api.go

```diff
@@ -969,11 +970,54 @@ func (a *PipedAPI) UpdateApplicationConfigurations(ctx context.Context, req *pip
 		}
 	}
 	for _, appInfo := range req.Applications {
+		// Retrieve env id by env name.
+		envID := ""
+		for key, value := range appInfo.Labels {
+			if key != config.EnvLabelKey {
+				continue
+			}
```

Comment:
Why not just use the `EnvName` field in the application info struct?

Reply:
Right